pkg/ddns/providers/shared/request.go

```diff
@@ -1,4 +1,4 @@
-package util
+package shared
```

Comment:
Why did you rename the package?
What is the difference between utils and shared? Is the package name sufficient and descriptive enough?

Reply:
Any specific meaningful name would conflict with other modules. Like request, for example. Therefore I have assigned the code snippet to a package below the providers which share this code. Alternative suggestions are welcome.

Reply:
Why is it necessary to export this 25-line function, which is only used in the provider package, to another package?